lib/lpc55-rot-startup/src/images.rs

```diff
@@ -120,20 +120,27 @@ impl Image {
         return true;
     }
 
-    // TODO: This is a particularly naive way to calculate the hash of the
-    // hubris image: https://github.com/oxidecomputer/hubris/issues/736
     pub fn get_hash(&self) -> [u8; 32] {
-        let img_ptr = self.get_img_start() as *const u8;
-        // The MPU requires 32 byte alignment and so the compiler pads the
-        // image accordingly. The length field from the image header does not
-        // (and should not) account for this padding so we must do that here.
-        let img_size = self.get_img_size().unwrap_lite() + 31 & !31;
-        let image = unsafe { core::slice::from_raw_parts(img_ptr, img_size) };
-
-        let mut img_hash = Sha3_256::new();
-        img_hash.update(image);
+        let mut hash = Sha3_256::new();
+
+        for start in IMAGE_FLASH.step_by(FLASH_PAGE_SIZE) {
+            let start: u32 = start.try_into().unwrap_lite();
+            if lpc55_romapi::validate_programmed(start, PAGE_SIZE) {
```

Comment:
Does this handle partially programmed pages correctly? I need to go review exactly how this would work on trailing pages (aside, I should sit down and finally rip out usage of this ROM API and just use the flash driver)

Reply:
> Does this handle partially programmed pages correctly?

I think so, assuming my reading of UM11126 § 8.6.7 is correct. This implementation includes the `0xff` trailing data added by the `write` command in the FWID hash.